compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt

```diff
@@ -173,7 +171,11 @@ class AdaptersProcessingStep(
     private fun generateConstructor(adapters: Set<AdapterKey>,
                                     genericTypes: List<TypeVariableName>): MethodSpec = MethodSpec.constructorBuilder()
             .addModifiers(Modifier.PUBLIC)
-            .addParameter(Moshi::class.java, "moshi")
+            .apply {
```

Comment:
Two good extensions that I use for JavaPoet and KotlinPoet: `applyIf(Boolean, T.() -> T): T` and `applyEach(Iterable<I>, T.(I) -> T): T`. We should probably add them to clean up this annoying double-nesting.

Reply:
Why would `applyEach`'s block return `T`, shouldn't it be `Unit`?

Reply:
Oops yeah they both do.

Reply:
Cool, just making sure. I added it to master.